test/src/main/java/ai/timefold/solver/test/api/score/stream/SingleConstraintAssertion.java

```diff
@@ -418,4 +418,366 @@ default void rewards() {
      */
     void rewards(String message);
 
+    /**
+     * Asserts that the {@link Constraint} being tested, given a set of facts,
+     * results in a specific penalty larger than given.
+     * <p>
+     * Ignores the constraint weight: it only asserts the match weights.
+     * For example:
+     * a match with a match weight of {@code 10} on a constraint with a constraint weight of {@code -2hard}
+     * reduces the score by {@code -20hard}.
+     * In that case, this assertion checks for {@code 10}.
+     * <p>
+     * An {@code int matchWeightTotal} automatically casts to {@code long} for {@link HardSoftLongScore long scores}.
+     *
+     * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
+     * @throws AssertionError when the expected penalty is not observed
+     */
+    default void penalizesByMoreThan(int matchWeightTotal) {
+        penalizesByMoreThan(null, matchWeightTotal);
+    }
+
+    /**
+     * As defined by {@link #penalizesByMoreThan(int)}.
+     *
+     * @param message sometimes null, description of the scenario being asserted
+     * @param matchWeightTotal at least 0, expected sum of match weights of matches of the constraint.
+     * @throws AssertionError when the expected penalty is not observed
+     */
+    void penalizesByMoreThan(String message, int matchWeightTotal);
```

Comment:
I wonder if we should have a `compareToGiven(Facts...).penalizesMore(String?)`

Example usage:

```java
constraintVerifier.verifyThat(MyConstraintProvider::fairnessConstraint)
    .given(Shift(Amy), Shift(Amy), Shift(Beth), Shift(Beth))
    .compareToGiven(Shift(Amy), Shift(Beth), Shift(Beth), Shift(Beth))
    .penalizesLess();
```